lib/public/components/Filters/RunsFilter/eorReasonsFilterModel.js

```diff
@@ -0,0 +1,119 @@
+ * or submit itself to any jurisdiction.
+ */
+
+import { RemoteData, Observable } from '/js/src/index.js';
+import { getRemoteData } from '../../../utilities/fetch/getRemoteData.js';
+
+const emptyEorReason = {
+    category: '',
+    title: '',
+    description: '',
+};
+
+/**
+ * Model storing state of a selection of run types picked from the list of all the existing run types
+ */
+export class EorReasonFilterModel extends Observable {
+    /**
+     * Constructor
+     */
+    constructor() {
+        super();
+        this._eorReasonTypes = RemoteData.notAsked();
+        this._fetchReasonTypes();
+        this._newEorReason = { category: emptyEorReason.category, title: emptyEorReason.title, description: emptyEorReason.description };
+    }
+
+    /**
+     * Retrieve a list of reason types from the API
+     *
+     * @returns {Promise<void>} resolves once the data has been fetched
+     */
+    async _fetchReasonTypes() {
+        this._eorReasonTypes = RemoteData.loading();
+        this.notify();
+
+        try {
+            const { data: reasonTypes } = await getRemoteData('/api/runs/reasonTypes');
+            this._eorReasonTypes = RemoteData.success(reasonTypes);
+        } catch (error) {
+            this._eorReasonTypes = RemoteData.failure(error);
+        }
+
+        this.notify();
+    }
+
+    /**
+     * Returns a boolean indicating whether this filter is considered empty
+     * @returns {boolean} indicating whether this filter is considered empty
+     */
+    isEmpty() {
+        return this._newEorReason.category == '' && this._newEorReason.title == '' && this._newEorReason.description == '';
+    }
+
+    /**
+     * Empties the filter. this.isEmpty() evaluates to true.
+     * @returns {void}
+     */
+    reset() {
+        this._newEorReason = { category: emptyEorReason.category, title: emptyEorReason.title, description: emptyEorReason.description };
+    }
+
+    /**
+     * Gets an object of url params to value for the api query
+     * @returns {Object} url query parameter to parameter value
+     */
+    getFilterQueryParams() {
+        return {
+            ...!this.newEorReason.category == '' && { 'filter[eor][category]': this.newEorReason.category },
+            ...!this.newEorReason.title == '' && { 'filter[eor][title]': this.newEorReason.title },
+            ...!this.newEorReason.description == '' && { 'filter[eor][description]': this.newEorReason.description },
+        };
+    }
+
+    /**
+     * Getter for the EOR reason type
+     *
+     * @return {RemoteData} the EOR reason type
+     */
+    get eorReasonTypes() {
+        return this._eorReasonTypes;
+    }
+
+    /**
+     * Getter for the EOR reason types
+     * @param {Object} reasonTypes object representing the possible EOR reason types
+     * @return {void}
+     */
+    set eorReasonTypes(reasonTypes) {
+        this._eorReasonTypes = reasonTypes;
+    }
+
+    /**
+     * Getter for the EOR reason type
```

Comment:
Documentation needs to be adapted to the corresponding `getter`. In this particular case, it looks like it is the same as the one above
Moreover, I would recommend renaming `_newEorReason` to something more representative such as `_filterEorReason`